### DISCUSSION TITLE
Lipgloss doesn't split on whitespace

### DISCUSSION THREAD

**Discussion by CalebJohnHunt** [charmbracelet/lipgloss#217] [Q&A] [answered]
I'm just trying to use Lipgloss to limit my text's width like so:

```go
wrapper.Copy().Width(wrapper.GetWidth() - 10).Render(text)
```
I have a `var wrapper lipgloss.Style`, I copy it, set its width to its old width minus some constant, and render that text.
I expect it to always split on whitespace (when possible), but I am occasionally finding weirdness such as the following:
```txt
I use a Moonlander at home and I don't ever use the red thumb button or the bottom
one because of how uncomfortable it is (and having rewritten this now, I'm        
considering swapping the top right thumb button - return - and the middle left thu
mb                                                                                
button - backspace - because i use backspace so much more).                       
```
(NB trailing spaces have been left in for accuracy)
The split between "I'm" and "considering" is totally reasonable, but what I don't understand is the split between "thu" and "mb" and newline after "mb". Since "...and the middle left thumb" is too long for the width (which is correct), I would instead expect
```txt
I use a Moonlander at home and I don't ever use the red thumb button or the bottom
one because of how uncomfortable it is (and having rewritten this now, I'm        
considering swapping the top right thumb button - return - and the middle left    
thumb button - backspace - because i use backspace so much more).                 
```
And to be clear, Lipgloss often _is_ splitting on the whitespace, but just rarely (about 1 in 15 cases for me) it splits in the middle of a word.

Any thoughts why? Am I missing a setting?

**Additional info** (will be updated if more is brought up):
+ Lipgloss version: v0.8.0 (though I was finding the same thing on older version)
* Go version: `go1.21.0  linux/amd64`

**Comment by beltagyy** [comment]
The mid-word splitting you are seeing is a known edge case in lipgloss word wrapping when the input contains long runs without spaces or when ANSI sequences change the measured width.\n\nA few things to try:\n1. Ensure your text is plain strings without hidden ANSI escapes before wrapping. Lipgloss measures rune width, and unexpected escape sequences can throw off the wrap point calculations.\n2. If you are constructing styles with borders or padding, apply Width on the inner text style rather than the outer bordered style so the calculation happens before borders eat space.\n3. For predictable wrapping, preprocess the text with a dedicated word-wrap library (e.g., muesli/reflow wordwrap) and pass the already-wrapped block into lipgloss for coloring and borders.\n\nIf you can reproduce this with a minimal snippet and no ANSI codes, opening an issue with the exact input string would help the maintainers fix the wrap logic.

**Reply by CalebJohnHunt** [reply, marked as answer]
Wow, my question was very poorly worded. I'm not sure why I didn't provide a repro.

**This issue has since been resolved.** After updating to v1.1.0, everything works as expected.

I will close this discussion (or mark it answered or whatever). Regardless, for future historians, here's a minimal repro. No idea who would want this, but have at it.

`go.mod`
```go.mod
module main

go 1.21.0

require github.com/charmbracelet/lipgloss v0.8.0

require (
	github.com/aymanbagabas/go-osc52/v2 v2.0.1 // indirect
	github.com/lucasb-eyer/go-colorful v1.2.0 // indirect
	github.com/mattn/go-isatty v0.0.18 // indirect
	github.com/mattn/go-runewidth v0.0.14 // indirect
	github.com/muesli/reflow v0.3.0 // indirect
	github.com/muesli/termenv v0.15.2 // indirect
	github.com/rivo/uniseg v0.2.0 // indirect
	golang.org/x/sys v0.7.0 // indirect
)
```
`main.go`
```golang
package main

import (
	"fmt"

	"github.com/charmbracelet/lipgloss"
)

func main() {
	style := lipgloss.NewStyle().Width(40)
	fmt.Println(style.Render(`abcdef - abcdef - abcdef - abcdef - abcdef - abcdef`))
}
```

What I'd expect:
```
abcdef - abcdef - abcdef - abcdef -
abcdef - abcdef
```
What I get:
```
abcdef - abcdef - abcdef - abcdef - abcd
ef -
abcdef
```
